packages/contracts/contracts/Deposit.sol

```diff
@@ -1,234 +1,281 @@
-    require(Math.max(_olderCheckpoint.checkpointedRange.start, _newerCheckpoint.checkpointedRange.start) < Math.min(_olderCheckpoint.checkpointedRange.end, _newerCheckpoint.checkpointedRange.end));
-
-    // Ensure that the plasma blocknumber of the olderCheckpoint is less than that of newerCheckpoint.
-    require(_olderCheckpoint.stateUpdate.plasmaBlockNumber < _newerCheckpoint.stateUpdate.plasmaBlockNumber);
-
-    // Ensure that the newerCheckpoint has no challenges.
-    bytes32 checkpointId = getCheckpointId(_newerCheckpoint.stateUpdate, _newerCheckpoint.stateUpdate.range);
-    require(checkpoints[checkpointId].outstandingChallenges == 0);
-    
-    // Ensure that the newerCheckpoint is no longer challengeable.
-    require(checkpoints[checkpointId].challengeableUntil < block.number + CHALLENGE_PERIOD);
-
-    // Delete the entries in exits and checkpoints at the olderCheckpointId
-    // delete exits
-    delete checkpoints[checkpointId];
-    delete exitsRedeemableAfter[checkpointId];
-  }
-
-  function challengeCheckpointInvalid(Challenge memory _challenge) private {
-    // TODO
-  }
-
-  function challengeLimboCheckpointAlternateTransaction(
-    bytes32 _limboCheckpoint,
-    bytes memory _alternateTransaction,
-    bytes memory _inclusionProof
-  ) private 
-  {
-    // TODO
-  }
-
-  function removeChallengeCheckpointInvalidHistory(Challenge memory _challenge) private {
-    // TODO 
-  }
-
-  function startExit(Challenge memory _checkpoint, bytes memory _witness) private {
-    // TODO
-  }
-
-  function challengeExitDeprecated(Challenge memory _checkpoint, bytes memory _transaction, bytes memory _inclusionProof) private {
-    // TODO
-  }
-
-  function finalizeExit(bytes32 _exit) private {
-    // TODO
-  }
-  
+     * @notice 
+     * @param _amount TODO
+     * @param _initialState  TODO
+     */
+    function deposit(uint256 _amount, dt.StateObject memory _initialState) public {
+        // Transfer tokens to the deposit contract
+        erc20.transferFrom(msg.sender, address(this), _amount);
+        // Create the Range, StateUpdate & Checkpoint
+        dt.Range memory depositRange = dt.Range({start:totalDeposited, end: totalDeposited + _amount });
+        dt.StateUpdate memory stateUpdate = dt.StateUpdate({
+            range: depositRange, stateObject: _initialState, 
+            depositAddress: address(this), plasmaBlockNumber: getLatestPlasmaBlockNumber() 
+        });
+        dt.Checkpoint memory checkpoint = dt.Checkpoint({
+            stateUpdate: stateUpdate,
+            subrange: depositRange
+        });
+        // Extend depositedRanges & increment totalDeposits
+        extendDepositedRanges(_amount);
+        // Calculate the checkpointId and add it to our finalized checkpoints
+        bytes32 checkpointId = getCheckpointId(checkpoint);
+        CheckpointStatus memory status = CheckpointStatus(
+            {challengeableUntil: block.number + CHALLENGE_PERIOD, outstandingChallenges: 0});
```

Comment:
No, you're correct, otherwise you'd have to download exclusion proofs--my suggestion in the spec was to do ``block.number - 1`` since we check for deletion with ``==0``.  Will change if that makes sense.  Great catch!